src/components/UserSelector.tsx

```diff
@@ -1,45 +1,39 @@
-import React, { useContext, useEffect, useState } from 'react';
+import React, { useEffect, useState } from 'react';
 import classNames from 'classnames';
-import { UserContext } from './UsersContext';
 import { User } from '../types/User';
+import { useAppDispatch, useAppSelector } from '../app/hooks';
+import { loadUsers } from '../features/users/usersSlice';
+import { authorSlice } from '../features/author/authorSlice';
 
 type Props = {
   value: User | null;
-  onChange: (user: User) => void;
 };
 
-export const UserSelector: React.FC<Props> = ({
-  // `value` and `onChange` are traditional names for the form field
-  // `selectedUser` represents what actually stored here
-  value: selectedUser,
-  onChange,
-}) => {
-  // `users` are loaded from the API, so for the performance reasons
-  // we load them once in the `UsersContext` when the `App` is opened
-  // and now we can easily reuse the `UserSelector` in any form
-  const users = useContext(UserContext);
+export const UserSelector: React.FC<Props> = ({ value: selectedUser }) => {
+  const { users } = useAppSelector(state => state.users);
+
+  const usersDispatch = useAppDispatch();
+
+  useEffect(() => {
+    usersDispatch(loadUsers());
+  }, [usersDispatch]);
+
   const [expanded, setExpanded] = useState(false);
 
   useEffect(() => {
     if (!expanded) {
       return;
     }
 
-    // we save a link to remove the listener later
     const handleDocumentClick = () => {
-      // we close the Dropdown on any click (inside or outside)
-      // So there is not need to check if we clicked inside the list
       setExpanded(false);
     };
 
     document.addEventListener('click', handleDocumentClick);
 
-    // eslint-disable-next-line consistent-return
     return () => {
       document.removeEventListener('click', handleDocumentClick);
     };
```

Comment:
The event listener for document clicks is added and removed based on the `expanded` state. While this works, it could lead to performance issues if the component is frequently re-rendered. Consider using a more efficient approach, such as attaching the listener once and using a ref to track the expanded state.